src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1447,26 +1344,12 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
       call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, 1)
       call mpas_pool_get_array(tracersSurfaceFluxPool, 'activeTracersSurfaceFlux', activeTracersSurfaceFlux)
       call mpas_pool_get_array(tracersSurfaceFluxPool, 'activeTracersSurfaceFluxRunoff', activeTracersSurfaceFluxRunoff)
-      ! allocate scratch space displaced density computation
-      call mpas_pool_get_field(scratchPool, 'densitySurfaceDisplaced', densitySurfaceDisplacedField)
-      call mpas_pool_get_field(scratchPool, 'thermalExpansionCoeff', thermalExpansionCoeffField)
-      call mpas_pool_get_field(scratchPool, 'salineContractionCoeff', salineContractionCoeffField)
-      call mpas_allocate_scratch_field(densitySurfaceDisplacedField, .true., .false.)
-      call mpas_allocate_scratch_field(thermalExpansionCoeffField, .true., .false.)
-      call mpas_allocate_scratch_field(salineContractionCoeffField, .true., .false.)
-
-      densitySurfaceDisplaced => densitySurfaceDisplacedField % array
-      thermalExpansionCoeff => thermalExpansionCoeffField % array
-      salineContractionCoeff => salineContractionCoeffField % array
+      call mpas_pool_get_array(diagnosticsPool, 'inSituThermalExpansionCoeff',thermalExpansionCoeff)
+      call mpas_pool_get_array(diagnosticsPool, 'inSituSalineContractionCoeff',salineContractionCoeff)
```

Comment:
@vanroekel I just retested, and this matches bfb with ocean/develop for all cases now. I see the EOS call below is deleted. Does this mean that `inSituThermalExpansionCoeff` and `inSituSalineContractionCoeff` can just be obtained because they were computed previously, presumably in the EOS call in diagnostics, so there is no need to calculate them again here?